src/org/opensolaris/opengrok/history/MercurialRepository.java

```diff
@@ -681,4 +681,43 @@ String determineParent() throws IOException {
 
         return executor.getOutputString().trim();
     }
+
+    @Override
+    String determineCurrentVersion() throws IOException {
+        String line = null;
+        File directory = new File(directoryName);
+
+        List<String> cmd = new ArrayList<>();
+        ensureCommand(CMD_PROPERTY_KEY, CMD_FALLBACK);
+        cmd.add(RepoCommand);
+        cmd.add("log");
+        cmd.add("-l");
+        cmd.add("1");
+        cmd.add("--template");
+        cmd.add("{date|isodate}: {node|short} {author} {desc|strip}");
+
+        ProcessBuilder pb = new ProcessBuilder(cmd);
+        pb.directory(directory);
+        Process process = pb.start();
+
+        try {
+            process.waitFor(15, TimeUnit.SECONDS);
```

Comment:
again , I though we deal with timeouts somehow globally, but @vladak knows this code better, otherwise seems ok, merging